api/controllers/album.js

```diff
@@ -38,3 +75,58 @@ export async function byAlbumID(req, res) {
     }
   }
 }
+/**
+ * function: getAlbumQuery
+ * creates query and parameters for an album query to the database
+ *
+ * @param {String} name - string for fuzzy search
+ * @param {String} tag - tag for fuzzy search
+ * @param {Int} parentID - parentID for exact search
+ * @param {String} date - updated after given date
+ * @param {String} keyword - keyword for fuzzy search
+ * @param {Int} albumID - albumID for exact search
+ *
+ * @return {Object[String, Array]} = template statement, parameters for template statement
+ */
+function getAlbumQuery(name, tag, parentID, date, keyword, albumID) {
+  const cols = `ID, Title, Parent, Tags, Keywords, Artists, Updated`;
+  const params = [];
+  let q = '';
+  const basicQuery = `SELECT ${cols} FROM Album WHERE 1=1`;
+  // parent
+  if (parentID) {
+    q += ` AND Album.Parent = ${parentID}`;
+    params.push(parentID);
+  }
+  // name
+  if (name) {
+    q += ` AND Album.Title LIKE '%${name}%'`;
+    params.push(name);
+  }
+  // tag
+  if (tag) {
+    q += ` AND Album.Tags LIKE '%${tag}%'`;
+    params.push(tag);
+  }
+  // date
+  if (date) {
+    q += ` AND Album.Updated > '${date}'`;
+    params.push(date);
+  }
+  // keyword
+  if (keyword) {
+    q += ` AND Album.Keywords LIKE '%${keyword}%'`;
+    params.push(keyword);
+  }
+  // album id
+  if (albumID) {
+    q += ` AND Album.ID = ${albumID}`;
+    params.push(albumID);
+  }
+  // // should we do this?
+  // else {
+  //   q += ` AND Parent IS NULL`;
+  // }
+  // return an object with a query and its parameters
+  return [`${basicQuery + q};`, params];
```

Comment:
better to return object instead of array. 
```
return { query:`${basicQuery + q};`, params };
```
then you can access 
`const { query, params } = getAlbumQuery(...)`.

That way order doesn't matter.